File: Interviews/GFG/Daily-Problems/2022/November/Day-20/LCM-Triplet.cpp
class Solution {
  public:
    long long lcmTriplets(long long N) {
        if (N % 6 == 0) {
            return (N - 1) * (N - 2) * (N - 3); 
        }
        if (N <= 2) {
            return N;
        }
        long long ans = N * (N - 1);
        for (long long i = N - 2, j = 2; i >= 1 and j < 3; i--) {
            if (__gcd(ans, i) == 1) {
                ans *= i;
                j += 1;
            }
        }
        
        return ans;
    }
};

File: Interviews/GFG/random-problems/Medium/LCM-Triplet/LCM-Triplet.py
import math


class Solution:
    def lcmTriplets(self, N):
        if N % 6 == 0:
            return (N - 1) * (N - 2) * (N - 3)
        if N <= 2:
            return N

        ans = N * (N - 1)
        for i in range(N - 2, 0, -1):
            if math.gcd(ans, i) == 1:
                ans *= i
                break

        return ans
